app/src/userpages/components/ShareDialog/ShareDialogContent/index.jsx

```diff
@@ -1,55 +1,180 @@
+import { addResourcePermission, saveUpdatedResourcePermissions } from '$userpages/modules/permission/actions'
+import Dialog from '$shared/components/Dialog'
+import useIsMountedRef from '$shared/utils/useIsMountedRef'
+import Buttons from '$shared/components/Buttons'
+
+import ShareDialogTabs, { type Tab } from '../ShareDialogTabs'
+import dialogStyles from '../shareDialog.pcss'
 
-import type { Permission, ResourceType, ResourceId } from '../../../flowtype/permission-types'
-import { addResourcePermission } from '../../../modules/permission/actions'
 import ShareDialogInputRow from './ShareDialogInputRow'
 import ShareDialogPermissionRow from './ShareDialogPermissionRow'
 import ShareDialogAnonymousAccessRow from './ShareDialogAnonymousAccessRow'
+import CopyLink from './CopyLink'
 
 import styles from './shareDialogContent.pcss'
 
+type StateProps = {
+    fetching: boolean,
+    anonymousPermission: ?Permission,
+}
+
 type DispatchProps = {
-    addPermission: (permission: Permission) => void
+    addPermission: (permission: Permission) => void,
+    save: () => Promise<void>,
 }
 
 type GivenProps = {
     resourceType: ResourceType,
     resourceId: ResourceId,
+    resourceTitle: string,
     addPermission: (permission: Permission) => {},
+    showEmbedInactiveWarning?: boolean,
+    clearShowEmbedInactiveWarning: Function,
+    onClose: () => void,
+    allowEmbed?: boolean,
+    setActiveTab: Function,
 }
 
-type Props = DispatchProps & GivenProps
+type Props = StateProps & DispatchProps & GivenProps
 
-export class ShareDialogContent extends Component<Props> {
-    render() {
-        return (
-            <div className={styles.container}>
-                <ShareDialogAnonymousAccessRow
-                    resourceType={this.props.resourceType}
-                    resourceId={this.props.resourceId}
-                />
-                <ShareDialogInputRow
-                    onAdd={(email: string) => this.props.addPermission({
-                        user: email,
-                        operation: 'read',
-                    })}
-                />
-                <ShareDialogPermissionRow
-                    resourceType={this.props.resourceType}
-                    resourceId={this.props.resourceId}
+export const ShareDialogContent = (props: Props) => {
+    const {
+        resourceType,
+        resourceId,
+        resourceTitle,
+        addPermission,
+        onClose,
+        allowEmbed,
+        save,
+        setActiveTab,
+        fetching,
+        anonymousPermission,
+    } = props
+
+    const [saving, setSaving] = useState(false)
+    const [showEmbedInactiveWarning, setShowEmbedInactiveWarning] = useState(false)
+    const isMountedRef = useIsMountedRef()
+
+    const dialogTitle = useMemo(() => (
+        !allowEmbed ? I18n.t('modal.shareResource.defaultTitle', {
+            resourceTitle,
+        }) : ''
+    ), [allowEmbed, resourceTitle])
```

Comment:
Does this need to be memoized? Not expensive computation and result is a string so it should maintain referential integrity.